tests/js-tests/src/MemoryModelTest/MemoryModelTest.js

```diff
@@ -0,0 +1,256 @@
+ Copyright (c) 2013-2014 Chukong Technologies Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+var memoryModelTestSceneIdx = -1;
+
+//------------------------------------------------------------------
+//
+// MemoryModelTestBase
+//
+//------------------------------------------------------------------
+var MemoryModelTestBase = BaseTestLayer.extend({
+    _title:"",
+    _subtitle:"",
+
+    onRestartCallback:function (sender) {
+        var s = new MemoryModelTestScene();
+        s.addChild(restartMemoryModelTest());
+        director.runScene(s);
+    },
+    onNextCallback:function (sender) {
+        var s = new MemoryModelTestScene();
+        s.addChild(nextMemoryModelTest());
+        director.runScene(s);
+    },
+    onBackCallback:function (sender) {
+        var s = new MemoryModelTestScene();
+        s.addChild(previousMemoryModelTest());
+        director.runScene(s);
+    },
+
+    // automation
+    numberOfPendingTests:function() {
+        return ( (arrayOfMemoryModelTest.length-1) - memoryModelTestSceneIdx );
+    },
+
+    getTestNumber:function() {
+        return memoryModelTestSceneIdx;
+    }
+
+});
+
+//------------------------------------------------------------------
+//
+// Set property on sprite
+//
+//------------------------------------------------------------------
+var SetPropertyMemoryModelTest = MemoryModelTestBase.extend({
+    _title:"Set Property Test",
+    _subtitle:"See console for possible errors",
+
+    ctor:function () {
+        this._super();
+        var sprite = new cc.Sprite(s_grossini_dance_atlas, cc.rect(0, 0, 85, 121));
+        var tag = 10;
+        this.addChild(sprite, 0, tag);
+        var x = winSize.width / 2;
+        var y = winSize.height / 2;
+        sprite.setPosition(x, y);
+
+        // add random property
+        sprite.randomProperty = "hello world";
+
+        sprite = this.getChildByTag(tag);
```

Comment:
Yes. probably we need to add that test as well. we still need to do a lot of tests.